src/java.base/share/classes/java/util/Properties.java

```diff
@@ -1515,7 +1515,7 @@ public synchronized Object computeIfAbsent(Object key,
 
     @Override
     public synchronized @Nullable Object compute(Object key,
-            BiFunction<? super Object, ? super Object, ?> remappingFunction) {
+            BiFunction<? super Object, ? super @Nullable Object, ?> remappingFunction) {
```

Comment:
The signature in `Map` is `BiFunction<? super K, ? super @Nullable V, ? extends @Nullable V>`. As you point out, `Properties` is a `Map<Object, Object>`.
So, to make that explicit, the signature here should be `BiFunction<? super K, ? super @Nullable Object, ? extends @Nullable Object>`.
We don't want to write `? extends @Nullable Object` to avoid an annotation on a bound that's not in bytecode (right?).
Just `?` is equivalent in this position anyways.
I just wanted to write out why the signature here looks slightly different from the version in `Map`.

Reply:
Yep, that's how I've been thinking about it.